mongo/mongo.go

```diff
@@ -27,3 +35,50 @@ func (m *Mongo) Init() (session *mgo.Session, err error) {
 	session.SetMode(mgo.Strong, true)
 	return session, nil
 }
+
+// GetRecipes retrieves all recipe documents from Mongo
+func (m *Mongo) GetRecipes() ([]recipe.Response, error) {
+	s := m.Session.Copy()
+	defer s.Close()
+
+	iter := s.DB(m.Database).C(m.Collection).Find(nil).Iter()
+	defer func() {
+		err := iter.Close()
+		if err != nil {
+			log.Event(context.Background(), "error closing iterator", log.ERROR, log.Error(err))
+		}
+	}()
+
+	results := []recipe.Response{}
+	if err := iter.All(&results); err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, errs.ErrRecipeNotFound
+		}
+		return nil, err
+	}
+	return results, nil
+}
+
+// GetRecipe retrieves a recipe document
+func (m *Mongo) GetRecipe(id string) (*recipe.Response, error) {
+	s := m.Session.Copy()
+	defer s.Close()
+	var recipe recipe.Response
+	err := s.DB(m.Database).C(m.Collection).Find(bson.M{"_id": id}).One(&recipe)
+	if err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, errs.ErrRecipeNotFound
+		}
+		return nil, err
+	}
+
+	return &recipe, nil
+}
+
+//AddRecipe adds a recipe document
+func (m *Mongo) AddRecipe(item recipe.Response) error {
```

Comment:
Is there any way to add multiple recipes in a single transaction in MongoDB? If it's easy I would suggest doing that to improve performance.

I had a look at the code in GlobalSign library and I don't think it's that trivial, so maybe this should be part of another task.

Reply:
Adding multiple recipes in a single transaction would be mentioned in an upcoming task when an UI would be created to do